### PR TITLE
Add support for platform binary to process exceptions

### PR DESCRIPTION
**Pull request by mlw**

### CODE REVIEW

File: Source/santad/DataLayer/WatchItemPolicy.h [hunk 1]
```diff
@@ -36,18 +37,22 @@ static constexpr bool kWatchItemPolicyDefaultAuditOnly = true;
...
 
     bool operator==(const Process &other) const {
       return binary_path == other.binary_path &&
              signing_id == other.signing_id && team_id == other.team_id &&
              cdhash == other.cdhash &&
-             certificate_sha256 == other.certificate_sha256;
+             certificate_sha256 == other.certificate_sha256 &&
+             platform_binary.has_value() == other.platform_binary.has_value() &&
+             platform_binary.value_or(false) ==
+                 other.platform_binary.value_or(false);
```
Comment (russellhancox) [inline Source/santad/DataLayer/WatchItemPolicy.h:55]: Am I reading correctly that this allows asserting in a policy that a binary is **not** a platform binary?

Comment (russellhancox) [reply Source/santad/DataLayer/WatchItemPolicy.h:55]: (I don't really mean that line specifically, but it was the easiest place to add the comment)

Comment (mlw) [reply Source/santad/DataLayer/WatchItemPolicy.h:55]: Yes, it does support that. At first I was only going to support `PlatformBinary=True`, but it seemed weird that `False` meant "don't check", so i added the 3rd state via the `optional`.

Comment (russellhancox) [reply Source/santad/DataLayer/WatchItemPolicy.h:55]: SGTM. I had started writing a comment about not needing the has_value condition if we can just treat unset as false but supporting explicit false makes sense, I can see uses for it.

**Review by russellhancox — APPROVED** [review]

**Pull request merged**